network-api/networkapi/wagtailpages/customblocks.py

```diff
@@ -209,3 +211,49 @@ class Meta:
         template = 'wagtailpages/blocks/quote_block.html'
         icon = 'openquote'
         help_text = 'Multiple quotes can be entered, but for now we are only using the first'
+
+
+class PulseProjectQueryValue(blocks.StructValue):
+    @property
+    def query(self):
+        # Replace any combination of spaces and commas with a single +
+        # because despite instructions to use spaces, someone, at some
+        # point, will accidentally use a comma, and that should be fine.
+        search_terms = self['search_terms'].strip()
+        query = re.sub(r'[\s,]+', '+', search_terms)
+        return query
+
+    @property
+    def max(self):
+        max_number_of_results = self['max_number_of_results']
+        return '' if max_number_of_results <= 0 else max_number_of_results
+
+
+class PulseProjectList(blocks.StructBlock):
+    search_terms = blocks.CharBlock(
+        help_text='Fill in any number of pulse entry search terms (separated by spaces).',
+    )
+
+    max_number_of_results = blocks.IntegerBlock(
+        min_value=0,
+        default=0,
+        required=False,
+        help_text='The maximum number of results to fetch (use 0 for "no maximum")',
+    )
+
+    reverse_chronological_listing = blocks.BooleanBlock(
+        default=False,
+        help_text='Check this box to select oldest entries first.',
```

Comment:
This is confusing. If a box labeled "reverse chronological listing" is checked, I'd expect to see newest first.